ts-github-profile-finder/src/components/ResultCase.tsx

```diff
@@ -0,0 +1,36 @@
+import PendingResult from "./PendingResult";
+import Result from "./Result";
+import RejectedResult from "./RejectedResult";
+
+import { UserInfo, UserInfoStatus } from "types";
+
+interface ResultCaseProps {
+  userInfo: UserInfo;
+  onUserInfoChange: (
+    targetUserInfo: UserInfo | null,
+    targetStatus: UserInfoStatus
+  ) => void;
+}
+
+const ResultCase = (props: ResultCaseProps) => {
+  const { userInfo, onUserInfoChange } = props;
+
+  switch (userInfo.status) {
+    case "pending":
+      return <PendingResult />;
+    case "resolved":
+      return (
+        <Result
+          userInfoData={userInfo.data}
+          onUserInfoChange={onUserInfoChange}
+        />
+      );
```

Comment:
위에 언급한 내용처럼 Result가 userInfoData를 null 없이 받게 한다면 이렇게! 

```suggestion
    case "resolved":
      return (
        userInfo !== null ?
        <Result
          userInfoData={userInfo.data}
          onUserInfoChange={onUserInfoChange}
        />
        : "what?"
      );
```

Reply:
prop drilling 할 때 저렇게 하는 게 훨 안전하겠구나!.~